### PR TITLE
[KEYCLOAK-2438] & [KEYCLOAK-3291]

### PR DESCRIPTION
**Pull request by abstractj**
[KEYCLOAK-2438] - Add display name to social login buttons
[KEYCLOAK-3291] - Names of social identity providers are wrongly capitalized (eg GitHub vs Github)

@stianst GitHub didn't allow me to reopen the previous PR (https://github.com/keycloak/keycloak/pull/3155) after my rebase.

All the tests failing were fixed, the only remaining thing, is to decide if you want to change https://github.com/keycloak/keycloak/compare/master...abstractj:KEYCLOAK-2438?expand=1#diff-e5482667296b532202309987b199206fR421. Other than that, I think we're done or at least close :)

### CODE REVIEW

File: testsuite/integration/src/test/resources/model/testrealm.json [hunk 1]
```diff
@@ -18,6 +18,7 @@
         {
             "providerId" : "google",
             "alias" : "google1",
+            "displayName": "Google",
```
Comment (stianst) [inline testsuite/integration/src/test/resources/model/testrealm.json]: Remove this

**Review by stianst — COMMENTED** [review]
Account management console is still showing lower case names (it's AccountFederatedIdentityBean).

Admin console identity providers list doesn't show display name for social providers or those that doesn't have a display name. In those cases it's not possible to click on the link in the display name. It's also not working properly when a specific provider is selected. The breadcrumb misses the value for name and the header is not showing proper name for social.

Finally, there's no tests around this. Can you please add that for login and acocunt mngmt at least?

**Comment by abstractj** [comment]
* [FIXED] - Account management console is still showing lower case names
* [FIXED] - The breadcrumb misses the value for name and the header is not showing proper name for social.
* [ADDED] - Finally, there's no tests around this. 

For "_Admin console identity providers list doesn't show display name for social providers or those that doesn't have a display name. In those cases it's not possible to click on the link in the display name. It's also not working properly when a specific provider is selected_". I couldn't reproduce your issue.

These are the screenshots after my changes:

- List of identity providers



- Editing Social



- Editing Custom



- Account page

**Comment by stianst** [comment]
Closing this PR as I've done a few tweaks and sent a new PR (https://github.com/keycloak/keycloak/pull/3268)

What I fixed was:
* Move changes to changelog for 2.3 (2.2 is released) and increased size of displayName to 255
* Use alias, not providerId, for fallback for custom providers with no display name
* Completely redone admin console parts as there was a few issues. Firstly it ended up sending the displayName in the realm rep, it used the wrong displayName (oidc ended up as saml) and also the display name wasn't updated on the edit idp page when editing the displayname/alias fields

**Pull request closed without merging**